diff -Naur ./external/siglist.c external/siglist.c
--- ./external/siglist.c	1999-01-07 13:43:15.000000000 -0500
+++ external/siglist.c	2012-12-17 12:33:24.000000000 -0500
@@ -18,7 +18,7 @@
 with Bash; see the file COPYING.  If not, write to the Free Software
 Foundation, 675 Mass Ave, Cambridge, MA 02139, USA. */
 
-#if defined(Linux) || defined(VM) || defined(MVS) || defined(VMS) || defined(OS9)
+#if defined(Linux) || defined(VM) || defined(MVS) || defined(VMS) || defined(OS9) || defined(Darwin)
 void initialize_siglist (){}
 #else
 
diff -Naur ./external/tcpsig.c external/tcpsig.c
--- ./external/tcpsig.c	1999-01-07 13:43:15.000000000 -0500
+++ external/tcpsig.c	2012-12-17 12:33:37.000000000 -0500
@@ -34,7 +34,7 @@
  * the system call 'sigignore' seems to behave as advertised.
  */
 
-#if defined(VM) || defined(MVS) || defined(VMS) || defined(OS9)
+#if defined(VM) || defined(MVS) || defined(VMS) || defined(OS9) || defined(Darwin)
 void init_signals(void){}
 #else
 
diff -Naur ./src/Makefile.Project src/Makefile.Project
--- ./src/Makefile.Project	1997-12-11 15:01:41.000000000 -0500
+++ src/Makefile.Project	2012-12-17 12:28:34.000000000 -0500
@@ -71,9 +71,9 @@
 vpath %.c $(VPATH)/../external
 vpath %.c $(VPATH)/../patterns
 
-CFLAGS_PROJECT = -I$(VPATH)/../include
+CFLAGS_PROJECT = -I$(VPATH)/../include -fPIC
 CXXFLAGS_PROJECT = $(CFLAGS_PROJECT)
-LDFLAGS_PROJECT = -L. -l$(LIBNICK)
+LDFLAGS_PROJECT = -L. -l$(LIBNICK) -fPIC -shared
 LDXXFLAGS_PROJECT = $(LDFLAGS_PROJECT)
 
 % : %.o
diff -Naur ./src/shmem.C src/shmem.C
--- ./src/shmem.C	1996-10-31 12:38:40.000000000 -0500
+++ src/shmem.C	2012-12-17 12:28:34.000000000 -0500
@@ -158,7 +158,7 @@
   void *data()       // user data pointer 
       { return wp(1); }
 
-  friend inline MemBlock *block(void *p) { return (MemBlock *) ((int *)p - 1); }
+  static inline MemBlock *block(void *p) { return (MemBlock *) ((int *)p - 1); }
 
   MemBlock *downF()  // pointer to the block, which is adjacent to down word
                      //  of free block
@@ -224,7 +224,7 @@
   s--;
   assert( s >= MINSZ );
   assert( s <= totsize - MINSZ );
-  MemBlock *q = block(p);
+  MemBlock *q = MemBlock::block(p);
   int size = q->size();
   int sz = size;
   if( sz < 0 ) sz = -sz;
@@ -315,7 +315,7 @@
   if( !vp )
     return;
   checkptr(vp,"checkptr free");
-  MemBlock *p = block(vp);
+  MemBlock *p = MemBlock::block(vp);
   int size = - p->size();
   assert( size > 0 );
   MemBlock *down = p->downU();
@@ -418,7 +418,7 @@
 void sh_cut(void *vp, int newsize)
   {
   checkptr(vp,"checkptr cut");
-  MemBlock *p = block(vp);
+  MemBlock *p = MemBlock::block(vp);
   int size = - p->size();
   assert( size > 0 );
   MemBlock *down = p->downU();
